lib/battle_boats/board.rb

```diff
@@ -16,11 +16,30 @@ def place_ships_randomly
         until ship_deployed?(ship: ship)
           coordinate = get_random_coordinate
           orientation = %i[horizontal vertical].sample
-          if orientation == :horizontal
-            place_ship_horizontally(coordinate: coordinate, ship: ship)
-          elsif orientation == :vertical
-            place_ship_vertically(coordinate: coordinate, ship: ship)
-          end
+          attempt_to_deploy_ship(ship: ship, coordinate: coordinate, orientation: orientation)
+        end
+      end
+    end
+
+    def ship_deployed?(ship:)
+      play_area.flatten.map(&:occupant).include?(ship)
+    end
+
+    def attempt_to_deploy_ship(ship:, coordinate:, orientation:)
```

Comment:
Does this method contain too many responsibilities? Is there an arrangement of this code that could maximize its readability?